### PR TITLE
Fix passive homes

### PR DESCRIPTION
**Pull request by julie-vienne**
Issue #28

### CODE REVIEW

**Comment by julie-vienne** [comment]
@floracharbo your comment did solve this specific problem, thanks.

However, I've noticed another problem if `n_homes = 2, n_homesP = 3`:

```
self.max_charge_t = np.where(last_step and self.avail_car, self.store0, self.cap)
```
as `self.store0` has the dimension of `n_homes` and `self.cap` has the dimension of `n_homesP`.

If they're the same, e.g. both 2, then comes another problem with the following assertion:

`assert abs(reward + res['hourly_total_costs'][time_step]) < 5e-3`

I thought it would be nice to directly include the passive houses in the reactive power PR but I will write my equations to set the reactive power of passive houses to zero until this issue is solved.

**Comment by floracharbo** [comment]
> harge_t = np.where(last_step and self.avail_car, self.store0, self.cap)

yes, we need to switch using the `set_passive_active` method when using the Battery methods to analyse the cars of the active of passive houses.

File: src/simulations/hedge.py
Comment (floracharbo) [inline src/simulations/hedge.py:55]: @julie-vienne fyi this was the main problem, that there there was a shallow copy instead of a deep copy, so when we edited the car info in the `passive_hedge` this also edited the main `prm`

**Review by floracharbo — COMMENTED** [review]
changes reviewed, now we just need to solve conflicts with main

**Comment by floracharbo** [comment]
note that I have debugged the passive homes with `manage_voltage = False`. I will merge at this point but I will leave it to you to implement it with the network modelling @julie-vienne.

File: src/simulations/hedge.py
Comment (julie-vienne) [reply src/simulations/hedge.py:55]: I see, good to know!

**Comment by julie-vienne** [comment]
Issue #35

**Review by floracharbo — APPROVED** [review]

**Comment by julie-vienne** [comment]
@floracharbo would you prefer that I open a new PR to work on #35?

**Comment by floracharbo** [comment]
> @floracharbo would you prefer that I open a new PR to work on #35?

I don't mind, if you reuse that branch please just wait until I have merged #31 (this may take a bit of time as I am waiting for tests to pass) and merge main into the branch again, so you are not too behind on the changes. Or it might just be easier to create a new branch from main.

**Comment by julie-vienne** [comment]
@floracharbo would it be okay for you if I tackle issue #35 with the reactive power issue as they are closely related?

**Comment by julie-vienne** [comment]
@floracharbo I think we can actually close this issue as we've seen in #29 that the passive homes now work when `manage_voltage = False` and `manage_agg_power = True`, I think we can solve it for the voltage management directly when we add the reactive power.

**Comment by floracharbo** [comment]
Yes, this is already closed. And I leave you to tackle #35 with the power flow modelling.

**Pull request merged**
